cmd/gardenlet/app/gardenlet.go

```diff
@@ -353,6 +357,12 @@ func NewGardenlet(ctx context.Context, cfg *config.GardenletConfiguration) (*Gar
 		return nil, errors.New("unable to extract Gardener`s cluster identity from cluster-identity ConfigMap")
 	}
 
+	// create the certificate manager to schedule certificate rotations
+	var certificateManager *certificate.Manager
+	if cfg.GardenClientConnection.KubeconfigSecret != nil {
+		certificateManager = certificate.NewCertificateManager(cfg.GardenClientConnection, bootstraputil.GetSeedName(cfg.SeedConfig), clientMap)
```

Comment:
I'm afraid you can't continue with the certificate manager in case `bootstraputil.GetSeedName(cfg.SeedConfig)` return `<ambiguous>` because it won't find a seed client with this name. WDYT?

Reply:
It should try to register a seed with the name `<ambiguous>` and then rotate the certs for the seed with this name.
I will test it.

Or shall we enforce that cfg.SeedConfig.name is set to prevent mistakes?

Reply:
IIUC, it'll not try to register a seed with that name (also <ambiguous> is not a valid K8s object name) but <ambiguous> is used for the CN part of the certificate.

Reply:
You are right. The CN part of the cert will be `gardener.cloud:system:seed:` and the seed registration will fail.
The gardenlet now exits the bootstrap process when the `seedConfig.name` is not set - so a wrong cert will not be requested. We are on the safe side.

If the configuration contains an invalid Seed configuration, the Seed registration will fail. This is true when using the bootstrap process or another provided kubeconfig.

Reply:
Please check an let me know if this is fine with you.

Reply:
I'm not sure if this has ever been about the use case if `seedConfig.name` is not set but rather if `seedConfig` is not set and you still get a certificate but with CN `gardener.cloud:system:seed:<ambiguous>`. Please correct me if I'm wrong /cc @rfranzke.

Reply:
If `seedConfig` is not set, then the `seedSelector` has to be set. 
If the `seedSelector` is set, then the gardenlet reconciles Shoots for multiple seeds. 
Therefore the CN of `gardener.cloud:system:seed:<ambiguous>` is correct in my opinion. 
The RBAC roles are associated with the Organization field in the Certificate anyways `O=gardener.cloud:system:seeds` - so having the aforementioned CN is not a problem in my opinion.

Reply:
Are you referring to [this code here](https://github.com/gardener/gardener/blob/ef0f4860c6b2c92e0752d4dbd7974a1cdc343ca2/pkg/gardenlet/bootstrap/certificate/certificate_rotation.go#L152) - yes I am afraid this will not work if the `seedConfig` is not set.

Reply:
> Are you referring to this code here? - yes I am afraid this will not work if the seedConfig is not set.
I am testing it atm.

Yes, that was my point.

Reply:
Looks good, thanks for taking care.